enclave-runner/src/usercalls/mod.rs

```diff
@@ -340,16 +374,181 @@ impl EnclaveState {
+                                },
+                                EnclaveEntry::ExecutableMain => {}
+                            }
+                            Ok((0,0))
+                        },
+                        Err(EnclaveAbort::IndefiniteWait) => Err(EnclaveAbort::IndefiniteWait),
+                        Err(EnclaveAbort::InvalidUsercall(n)) => Err(EnclaveAbort::InvalidUsercall(n)),
+                        Err(EnclaveAbort::MainReturned) => Err(EnclaveAbort::MainReturned),
+                        Err(EnclaveAbort::Secondary) => Err(EnclaveAbort::Secondary),
+                    }
+                },
+                CoResult::Return((tcs,v1, v2)) => {
+                    match state.mode {
+                        EnclaveEntry::Library => {
+                            self.threads_queue.push(
+                                StoppedTcs {
+                                    tcs,
+                                    event_queue: state.event_queue
+                                });
+                            Ok((v1, v2))
+                        }
+                        EnclaveEntry::ExecutableMain => {
+                            return Err(EnclaveAbort::MainReturned);
+                        }
+                        EnclaveEntry::ExecutableNonMain => {
+                            assert_eq!(
+                                (v1, v2),
+                                (0, 0),
+                                "Expected enclave thread entrypoint to return zero"
+                            );
+                            // If the enclave is in the exit-state, threads are no
+                            // longer able to be launched
+                            if !self.exiting.load(Ordering::SeqCst) {
+                                self.threads_queue.push(StoppedTcs {
+                                    tcs,
+                                    event_queue: state.event_queue,
+                                });
+                            }
+                            Ok((0,0))
+                        }
+                    }
+                }
+            };
+
+            match (my_result, state.mode ) {
+                (e, EnclaveEntry::Library)
+                | (e, EnclaveEntry::ExecutableMain)
+                | (e @ Err(EnclaveAbort::Secondary), EnclaveEntry::ExecutableNonMain)
+                => { return e },
+                (Ok(_), EnclaveEntry::ExecutableNonMain) => {
+                    continue;
+                },
+                (Err(e @ EnclaveAbort::Exit {..}), EnclaveEntry::ExecutableNonMain)
+                | (Err(e @ EnclaveAbort::InvalidUsercall(_)), EnclaveEntry::ExecutableNonMain)
+                => {
+                    let cmd = self.kind.as_command().unwrap();
+                    let mut cmddata = cmd.data.lock().unwrap();
+
+                    if cmddata.primary_panic_reason.is_none() {
+                        cmddata.primary_panic_reason = Some(e)
```

Comment:
Somewhere, we lost the logic where this gets set for `EnclaveEntry::ExecutableMain`

Reply:
According to the previous architecture, this was only set for threads. 
I tried to mirror the error handling as it was before, hence, it isn't set for `EnclaveEntry::ExecutableMain`.

Is this variable not meant for understanding what the primary reason might be if a secondary thread panics?

Reply:
Yes, you are correct. Thanks for paying attention!